File: typescript/darts/darts.ts
export function score(x: number, y: number): number {
  const distance = Math.sqrt(x * x + y * y);
  if (distance <= 1.0) return 10;
  if (distance <= 5.0) return 5;
  if (distance <= 10.0) return 1;
  return 0;
}

File: r/darts/darts.R
score <- function(x, y) {
  distance <- sqrt(x^2 + y^2)
  if (distance <= 1.0) { 
    10
  } else if (distance <= 5.0) { 
    5
  } else if (distance <= 10.0) { 
    1
  } else {
    0
  }
}
